spec/unit/models/project_spec.rb

```diff
@@ -49,6 +61,80 @@
           .to raise_error(/Cannot transfer output stage from foo to bar/)
       end
     end
+
+    it 'creates output stage with the same prefix as in the original project' do
+      expect(GoodData::AdsOutputStage).to receive(:create)
+        .with(hash_including(output_stage_prefix: original_prefix))
+      subject.transfer_output_stage(from_project, to_project, {})
+    end
+
+    context 'when ads_output_stage_prefix is specified' do
+      let(:options) do
+        { ads_output_stage_prefix: prefix }
+      end
+
+      it 'creates output stage with the specified prefix' do
+        expect(GoodData::AdsOutputStage).to receive(:create)
+          .with(hash_including(output_stage_prefix: prefix))
+        subject.transfer_output_stage(from_project, to_project, options)
+      end
+    end
+
+    context 'when ads_output_stage_uri is specified' do
+      let(:options) do
+        { ads_output_stage_uri: stage_uri }
+      end
+
+      it 'creates output stage with the specified output stage uri' do
+        expect(GoodData::AdsOutputStage).to receive(:create)
+          .with(hash_including(ads: stage_uri))
+        subject.transfer_output_stage(from_project, to_project, options)
+      end
+    end
+
+    context 'when add process already exists' do
+      let(:to_add) { double('add') }
+      let(:to_output_stage) { double(GoodData::AdsOutputStage) }
+
+      before do
+        allow(to_project).to receive(:processes).and_return([process])
+        allow(to_project).to receive(:add).and_return(to_add)
+        allow(to_output_stage).to receive(:schema)
+        allow(to_add).to receive(:output_stage).and_return(to_output_stage)
+        allow(to_output_stage).to receive(:schema=)
+        allow(to_output_stage).to receive(:output_stage_prefix=)
+      end
+
+      it 'sets output stage prefix to the same value as in the original project' do
+        expect(to_output_stage).to receive(:output_stage_prefix=).with(original_prefix)
+        expect(to_output_stage).to receive(:save)
+        subject.transfer_output_stage(from_project, to_project, {})
+      end
+
+      context 'when ads_output_stage_prefix is specified' do
```

Comment:
warning: Similar code found in project_spec.rb:114, project_spec.rb:126 (mass = 62) (flay)